pkg/cvo/sync_worker.go

```diff
@@ -227,7 +227,7 @@ func (w *SyncWorker) StatusCh() <-chan SyncWorkerStatus {
 func (w *SyncWorker) syncPayload(ctx context.Context, work *SyncWork,
 	reporter StatusReporter) ([]configv1.ClusterVersionCapability, error) {
```

Comment:
We should grow godocs here, so the semantics of the returned caps list are explicitly declared.  The returned caps list is from a313095a1c3 (#758), and gets passed up through `loadUpdatedPayload` to `Update` where it is fed into `SetFromImplicitlyEnabledCapabilities`.  But thinking about it now, I'm a bit confused by the semantics.  Wouldn't we actually want to feed the set of enabled capabilities (both implicit and explicit), to be compared with the set of requested caps, to determine the new set of enabled capabilities?  For example:

1. Admin requests cap1 and cap2.  Enabled set is cap1 and cap2.  Implicitly enabled set is empty.
2. Admin stops requesting cap1.  Enabled set is cap1 and cap2.  Implicitly enabled set is cap2.
3. Admin requests cap1 again, but stops requesting cap2.  Enabled set is cap1 and cap2.  Implicitly enabled set is cap1.

I'm worried that feeding just the currently-implicit-set into `SetFromImplicitlyEnabledCapabilities` will flub step 3 and miss the fact that cap1 needs to stay enabled.

Reply:
The logic to figure out "1. 2. 3." is in `GetImplicitlyEnabledCapabilities` which does take the entire Capabilities structure. Per its godoc, `SetFromImplicitlyEnabledCapabilities`, given implicitly enabled capabilities which were already determined by `GetImplicitlyEnabledCapabilities` simply "...updates ... with the given implicitly enabled capabilities and ensures each is in the enabled map".

Reply:
ACK to the godoc for this method and a unit test.